tests/validators/test_dict.py

```diff
@@ -21,24 +22,29 @@ def test_dict(py_and_json: PyAndJson):
         v.validate_test([])
 
 
+@dataclass
+class Foobar:
+    x = 1
+
+
 @pytest.mark.parametrize(
     'input_value,expected',
     [
-        ({'1': b'1', '2': b'2'}, {'1': '1', '2': '2'}),
-        (OrderedDict(a=b'1', b='2'), {'a': '1', 'b': '2'}),
+        ({'1': b'1', '2': b'2'}, {'1': 1, '2': 2}),
+        (OrderedDict(a=b'1', b='2'), {'a': 1, 'b': 2}),
         ({}, {}),
         ('foobar', Err("Input should be a valid dictionary [type=dict_type, input_value='foobar', input_type=str]")),
-        ([], Err('Input should be a valid dictionary [type=dict_type,')),
-        ([('x', 'y')], Err('Input should be a valid dictionary [type=dict_type,')),
-        ([('x', 'y'), ('z', 'z')], Err('Input should be a valid dictionary [type=dict_type,')),
-        ((), Err('Input should be a valid dictionary [type=dict_type,')),
-        ((('x', 'y'),), Err('Input should be a valid dictionary [type=dict_type,')),
-        ((type('Foobar', (), {'x': 1})()), Err('Input should be a valid dictionary [type=dict_type,')),
+        ([], {}),
+        ([('x', '1')], {'x': 1}),
+        ([('x', '1'), ('z', b'2')], {'x': 1, 'z': 2}),
+        ((), {}),
+        ((('x', '1'),), {'x': 1}),
```

Comment:
These are all accepted now. I also changed the value type to be `int` so that we know we're not swapping keys and values validators.